app/controllers/users_controller.rb

```diff
@@ -26,10 +27,30 @@ def create
       render 'new'
     end
   end
+
+  def new_with_invitation_token 
+    invitation = Invitation.where(token: params[:token]).first
```

Comment:
Yes, `where().first` orders the collection while `where().take` does not and takes first record immediately, what makes it more efficient.

Good job!